File: Commons/Action/src/main/java/com/greentree/commons/action/observer/run/RunAction.java
package com.greentree.commons.action.observer.run;

import com.greentree.commons.action.MultiAction;

public class RunAction extends MultiAction<Runnable> implements IRunnableAction {
	private static final long serialVersionUID = 1L;

	@Override
	public void event() {
		for(var l : listeners)
			l.run();
	}

}

File: Commons/Action/src/main/java/com/greentree/commons/action/observer/run/RunAction.kt
package com.greentree.commons.action.observer.run

import com.greentree.commons.action.MultiAction

open class RunAction : MultiAction<Runnable>(), IRunnableAction {

	override fun event() {
		for(l in listeners)
			l.run()
	}

	companion object {

		private const val serialVersionUID = 1L
	}
}
